### DISCUSSION TITLE
Is it possible to maximise the amount of jobs per tour?

### DISCUSSION THREAD

**Discussion by williamrijksen** [reinterpretcat/vrp#34] [Q&A] [answered]
Is it possible to maximise the amount of jobs per tour?

**Comment by reinterpretcat** [comment]
What would be the difference from the objective which maximizes total amount of assigned jobs and then minimizes amount of tours?

**Reply by williamrijksen** [reply]
Oh. Let me explain it a little bit more. Question is to set a limit on the deliveries per tour.

**Reply by reinterpretcat** [reply]
Have you tried to use `limits.tourSize` which limits amount of activities (all types, incl. breaks, etc.)? It is mentioned here: https://reinterpretcat.github.io/vrp/concepts/pragmatic/problem/vehicles.html

**Reply by williamrijksen** [reply]
Thanks, I'll give it a try later today!

**Reply by williamrijksen** [reply]
I looked at it, but it's not exactly what I want. We want to force reload always after max three deliveries. A tour includes also reloads so it's resulting in constraint errors. 

Is another feature available to make this happen?

**Reply by reinterpretcat** [reply]
As replied in another topic, it sounds for me like a setting demand of each job as one and capacity of each vehicle to 3 should work. If jobs have some other sizes which might fit or not to the vehicle, then you can use multidimensional array as capacity/demand, e.g. capacity: [3, 100], demand of some jobs: [1, 60], [1, 20], [1, 40]. Here the second dimension is used to define specific size, the first - limits amount of jobs served per vehicle

**Reply by williamrijksen** [reply]
Nice, I looked at it, but it rans into issues when setting the tourSize to 3 clients and expect the vehicle to reload, even when the capacity is not full. Is it possible that the `limits.tourSize` only works when no reloads are applied?

**Reply by williamrijksen** [reply]
I think I need a property called `tripSize`, which limits the amount of deliveries between the reloads. Does the library already supply this functionality?

**Reply by reinterpretcat** [reply]
`tourSize` counts all activities except departure/arrival, so you can't use it to limit trips between reload. There is no built-in `tripSize` constraint at the moment.

**Reply by williamrijksen** [reply]
> `tourSize` counts all activities except departure/arrival, so you can't use it to limit trips between reload. There is no built-in `tripSize` constraint at the moment.

Thanks for your reply. How difficult will it be implementing it? Eventually I can create a PR myself to implement it

**Reply by reinterpretcat** [reply]
It is relatively simple to do with trivial approach, but simple implementation will affect performance as you would need to count how many jobs are served between different reloads. This approach brings an additional O(N) complexity for each job insertion evaluation. Internally, there are some caching approach used and some cached values can be reused (e.g. RELOAD_INTERVALS_KEY), but this would require a bit of understanding of internal details which are not well described at the moment (only as doc comments).

You can check `tourSize` as a reference for simple implementation: https://github.com/reinterpretcat/vrp/blob/master/vrp-core/src/construction/constraints/tour_size.rs

**Reply by williamrijksen** [reply]
Okay nice, I'll take a look this weekend. Of course it brings indeed  an additional O(N) complexity, but I think it's not a big deal for my use case because it has not that much data.

**Comment by williamrijksen** [comment, marked as answer]
Found a solution which fits my case: I added a second item in the capacity array with the max amount of deliveries. In the demand I added a second item with `1`. So on every delivery the load decreases until the max amount of deliveries is reached.
